### PR TITLE
feat(processor.extensions) Add support for CPU flag detection

### PR DESCRIPTION
**Pull request by jcpunk**
This should add in an extra set of attributes for finding your microarch or CPU extensions.

Addresses #2663

I'll probably need help with the tests...

### CODE REVIEW

File: spec/facter/resolvers/processors_spec.rb [hunk 1]
```diff
@@ -21,6 +21,9 @@
       end
 
       let(:speed) { 2_294_000_000 }
+      let(:extensions) do
+        ['x86_64', 'x86_64-v1', 'x86_64-v2', 'x86_64-v3']
```
Comment (joshcooper) [inline spec/facter/resolvers/processors_spec.rb:25]:
```suggestion
        %w[x86_64 x86_64-v1 x86_64-v2 x86_64-v3]
```

File: spec/facter/resolvers/processors_spec.rb [hunk 2]
```diff
@@ -45,6 +48,12 @@
 
         expect(result).to eq(speed)
       end
+
+      it 'returns extensions supported' do
+        result = Facter::Resolvers::Linux::Processors.resolve(:extensions)
```
Comment (joshcooper) [inline spec/facter/resolvers/processors_spec.rb:53]: Tests are failing with:

```
  1) Facter::Resolvers::Linux::Processors when on x86 architecture when cpuinfo file is readable returns extensions supported
     Failure/Error: expect(result).to eq(extensions)

       expected: ["x86_64", "x86_64-v1", "x86_64-v2", "x86_64-v3"]
            got: ["i386"]
```

You'll want to stub the call to `Facter::Resolvers::Uname.resolve(:processor)` so it returns the expected values.

Comment (jcpunk) [reply spec/facter/resolvers/processors_spec.rb:57]: I feel like the current crop of test failures isn't me...

**Comment by jcpunk** [comment]
Rebased off HEAD

File: spec/facter/resolvers/uname_spec.rb [hunk 3]
```diff
@@ -27,6 +27,10 @@
     Facter::Resolvers::Uname.invalidate_cache
   end
 
+  after do
+    Facter::Resolvers::Uname.invalidate_cache
```
Comment (joshcooper) [inline spec/facter/resolvers/uname_spec.rb:31]: suggest moving this into the the existing `after` block to fix the rubocop issue

**Review by joshcooper — APPROVED** [review]
```
$ bundle exec facter processors.extensions
[
  "x86_64",
  "x86_64-v1",
  "x86_64-v2",
  "x86_64-v3",
  "x86_64-v4"
]
```

**Comment by joshcooper** [comment]
Thanks @jcpunk !

**Pull request merged**